packages/wrangler/src/d1/insights.ts

```diff
@@ -0,0 +1,175 @@
+};
+
+type HandlerOptions = StrictYargsOptionsToInterface<typeof Options>;
+export const Handler = withConfig<HandlerOptions>(
+	async ({
+		name,
+		config,
+		json,
+		count,
+		timePeriod,
+		sortType,
+		sortBy,
+		sortDirection,
+	}): Promise<void> => {
+		const accountId = await requireAuth(config);
+		const db: Database = await getDatabaseByNameOrBinding(
+			config,
+			accountId,
+			name
+		);
+
+		const result = await getDatabaseInfoFromId(accountId, db.uuid);
+
+		const output: Record<string, string | number>[] = [];
+
+		if (result.version !== "alpha") {
+			const convertedTimePeriod = Number(timePeriod.replace("d", ""));
+			const endDate = new Date();
+			const startDate = new Date(
+				new Date(endDate).setDate(endDate.getDate() - convertedTimePeriod)
+			);
+			const parsedSortBy =
+				cliOptionToGraphQLOption[
+					sortBy as "time" | "reads" | "writes" | "count"
+				];
+			const orderByClause =
+				parsedSortBy === "count"
+					? `${parsedSortBy}_${sortDirection}`
+					: `${sortType}_${parsedSortBy}_${sortDirection}`;
+			const graphqlQueriesResult =
+				await fetchGraphqlResult<D1QueriesGraphQLResponse>({
+					method: "POST",
+					body: JSON.stringify({
+						query: `query getD1QueriesOverviewQuery($accountTag: string, $filter: ZoneWorkersRequestsFilter_InputObject) {
+								viewer {
+									accounts(filter: {accountTag: $accountTag}) {
+										d1QueriesAdaptiveGroups(limit: ${count}, filter: $filter, orderBy: [${orderByClause}]) {
+											sum {
+												queryDurationMs
+												rowsRead
+												rowsWritten
+											}
+											avg {
+												queryDurationMs
+												rowsRead
+												rowsWritten
+											}
+											count
+											dimensions {
+													query
+											}
+										}
+									}
+								}
+							}`,
+						operationName: "getD1QueriesOverviewQuery",
+						variables: {
+							accountTag: accountId,
+							filter: {
+								AND: [
+									{
+										datetimeHour_geq: startDate.toISOString(),
+										datetimeHour_leq: endDate.toISOString(),
+										databaseId: db.uuid,
+									},
+								],
+							},
+						},
+					}),
+					headers: {
+						"Content-Type": "application/json",
+					},
+				});
+
+			if (graphqlQueriesResult) {
+				graphqlQueriesResult.data?.viewer?.accounts[0]?.d1QueriesAdaptiveGroups?.forEach(
+					(row) => {
+						if (!row.dimensions.query) return;
+						output.push({
+							query: row.dimensions.query,
+							avgRowsRead: row?.avg?.rowsRead ?? 0,
+							totalRowsRead: row?.sum?.rowsRead ?? 0,
+							avgRowsWritten: row?.avg?.rowsWritten ?? 0,
+							totalRowsWritten: row?.sum?.rowsWritten ?? 0,
+							avgDurationMs: row?.avg?.queryDurationMs ?? 0,
+							totalDurationMs: row?.sum?.queryDurationMs ?? 0,
+							numberOfTimesRun: row?.count ?? 0,
+						});
+					}
+				);
+			}
+		}
+
+		if (json) {
+			logger.log(JSON.stringify(output, null, 2));
+		} else {
+			await printWranglerBanner();
```

Comment:
Should we also be calling `d1BetaWarning()` here too?

(Actually looking at the rest of the code, `d1BetaWarning()` is currently only used in the epilogue... But I wonder if it would be worth reiterating that warning on each function?)